.github/workflows/release.yaml

```
@@ -0,0 +1,150 @@
+        arch:
+        - cpu: x86_64
+          platform: amd64
+        os:
+        - name: linux
+          distro: debian
+          version: 11
+          codename: bullseye
+          pkg_type: deb
+        - name: linux
+          distro: ubuntu
+          version: 21.04
+          codename: hirsute
+          kernel: linux
+          pkg_type: deb
+        - name: linux
+          distro: centos
+          version: 7
+          codename: 7
+          pkg_type: rpm
+        - name: linux
+          distro: centos
+          version: 8
+          codename: 8
+          pkg_type: rpm
+        postgres:
+        - version: "12"
+        - version: "13"
+        - version: "14"
+
+    steps:
+    - name: Install PackageCloud
+      if: ${{ startsWith(github.ref, 'refs/tags') }}
+      run: |
+        sudo gem install package_cloud --no-doc
+
+    - uses: actions/checkout@v2
+
+    - name: Gather Metadata
+      id: metadata
+      run: |
+        if [[ "${GITHUB_REF}" =~ 'refs/tags' ]]; then \
+          version="$(echo "${GITHUB_REF}" | sed -e 's|refs/tags/||')"; \
+          tag="${version}"; \
+        else \
+          tag="$(cargo metadata --format-version 1 --no-deps | jq -r '.packages[0].version')"; \
+          version="${tag}-snapshot"; \
+        fi;
+        echo "::set-output name=version::${version}"
+        echo "::set-output name=tag::${tag}"
+        echo "::set-output name=filename::promscale_extension-${tag}.${{ matrix.os.distro }}${{ matrix.os.version }}.${{ matrix.arch.cpu }}.${{ matrix.os.pkg_type }}"
+        echo "::set-output name=outfile::promscale_extension-${version}.${{ matrix.os.distro }}${{ matrix.os.version }}.${{ matrix.arch.cpu }}.${{ matrix.os.pkg_type }}"
+        echo "::set-output name=image::promscale_extension:${tag}.${{ matrix.os.distro }}${{ matrix.os.version }}"
+
+    - name: Setup QEMU
+      uses: docker/setup-qemu-action@v1
+
+    - name: Setup Docker Buildx
+      uses: docker/setup-buildx-action@v1
+
+    - name: Build Package
+      uses: docker/build-push-action@v2
+      with:
+        context: .
+        load: true
+        platforms: ${{ matrix.os.name }}/${{ matrix.arch.platform }}
+        file: dist/${{ matrix.os.pkg_type }}.dockerfile
+        build-args: |
+          OS_NAME=${{ matrix.os.distro }}
+          OS_VERSION=${{ matrix.os.version }}
+          PG_VERSION=${{ matrix.postgres.version }}
+          RUST_VERSION=${{ env.RUST_VERSION }}
+        tags: ${{ steps.metadata.outputs.image }}
+
+    - name: Extract Package
+      env:
+        PLATFORM: ${{ matrix.os.name }}/${{ matrix.arch.platform }}
+      run: |
+        mkdir -p artifacts
+        container="$(docker create --platform "${PLATFORM}" "${{ steps.metadata.outputs.image }}")"
+        docker cp ${container}:/dist/${{ steps.metadata.outputs.filename }} artifacts/${{ steps.metadata.outputs.outfile }}
+        docker rm -f "${container}"
+        ls -la artifacts/*
+
+    - name: Upload Artifact for GitHub Release
```

Nit: this won't be associated with a release 
```suggestion
    - name: Upload Artifact for Job
```